### PR TITLE
POC Authentication in ElasticSearch

### PR DESCRIPTION
**Pull request by aduprat**
Hi,

Here's a POC about authentication on ElasticSearch.
_This PR doesn't aim to be merged._

**TODO**: 
* rework git history
* add unit tests
* have decent code (handle exceptions...)

### CODE REVIEW

**Comment by Arsnael** [comment]
Read it

File: dockerfiles/run/guice/cassandra/destination/conf/elasticsearch.properties [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
 elasticsearch.masterHost=elasticsearch
 elasticsearch.port=9200
+# Only http or https are accepted
+elasticsearch.hostScheme=https
+elasticsearch.user=elasticsearch
+elasticsearch.password=secret
```
Comment (chibenwa) [inline dockerfiles/run/guice/cassandra/destination/conf/elasticsearch.properties]: We may need to comment that by default...

File: backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java [hunk 2]
```diff
@@ -137,13 +164,39 @@ public static Builder builder() {
...
+    public static final String HTTPS_HOST_SCHEME = "https";
+
     public static final ElasticSearchConfiguration DEFAULT_CONFIGURATION = builder()
         .addHost(Host.from(LOCALHOST, DEFAULT_PORT))
         .build();
 
     public static ElasticSearchConfiguration fromProperties(Configuration configuration) throws ConfigurationException {
         return builder()
             .addHosts(getHosts(configuration))
+            .hostScheme(HostScheme.from(configuration.getString(ELASTICSEARCH_HOST_SCHEME, HostScheme.DEFAULT.scheme)))
```
Comment (chibenwa) [inline backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java]: We are duplicating the default value behavior here for hostScheme

**Comment by Arsnael** [comment]
CI restart...

test this please

File: backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java [hunk 3]
```diff
@@ -137,13 +164,39 @@ public static Builder builder() {
...
+            for (HostScheme hostScheme : values()) {
+                if (hostScheme.scheme.equalsIgnoreCase(scheme)) {
+                    return hostScheme;
+                }
+            }
+            throw new RuntimeException("Unkown host scheme");
+        }
+    }
+
+    public static final String HTTPS_HOST_SCHEME = "https";
```
Comment (trantienduchn) [inline backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java]: I guess you can remove it

File: backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java [hunk 4]
```diff
@@ -137,13 +164,39 @@ public static Builder builder() {
...
+        public static final HostScheme DEFAULT = HTTP;
+
+        private final String scheme;
+
+        HostScheme(String scheme) {
+            this.scheme = scheme;
+        }
+
+        public static HostScheme from(String scheme) {
+            for (HostScheme hostScheme : values()) {
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchConfiguration.java]: `TaskManager.Status.fromString()` can be an example

File: dockerfiles/run/guice/cassandra/destination/conf/elasticsearch.properties (hunk 1 above)
Comment (trantienduchn) [reply dockerfiles/run/guice/cassandra/destination/conf/elasticsearch.properties]: Should it be added in other elasticsearch.properties template files?

**Comment by trantienduchn** [comment]
see the complete version #2832

**Pull request closed without merging**